Tests/ML/utils/test_io_util.py

```diff
@@ -198,76 +198,110 @@ def write_test_dicom(array: np.ndarray, path: Path) -> None:
     This function DOES NOT create a usable Dicom file and is meant only for testing: tags are set to
     random/default values so that pydicom does not complain when reading the file.
     """
-
     image = sitk.GetImageFromArray(array)
     writer = sitk.ImageFileWriter()
-    writer.KeepOriginalImageUIDOn()
     writer.SetFileName(str(path))
     writer.Execute(image)
 
 
-def test_load_dicom_image_ones(test_output_dirs: TestOutputDirectories) -> None:
+def get_mock_function(monochrome2: bool, bits_stored: Optional[int] = None) -> Callable:
     """
-    Test loading of 2D Dicom images filled with (uint16) ones.
+    SimpleITK does not allow us to set the Photometric Interpretation and Stored Bits tags when writing the Dicom image.
+    In these tests, if the image should be MONOCHROME1 we write an inverted image with tag MONOCHROME2
+    and use this wrapper around the SimpleITK metadata reader to make it look to the test like the tag was MONOCHROME1.
+    Similarly, we write images with StoredBits set to 16, but use this wrapper to change StoredBits while reading.
     """
-    array_size = (20, 30)
-    array = np.ones(array_size, dtype='uint16')
-    array[::2] = 0
-    assert array.shape == array_size
+    get_metadata_function = sitk.ImageFileReader.GetMetaData
 
-    dcm_file = Path(test_output_dirs.root_dir) / "file.dcm"
-    assert is_dicom_file_path(dcm_file)
-    write_test_dicom(array, dcm_file)
-
-    image = load_dicom_image(dcm_file)
-    assert image.ndim == 3 and image.shape == (1, ) + array_size
-    assert np.array_equal(image, array[None, ...])
+    def mock_function(image_reader: sitk.ImageFileReader, key: str) -> str:
+        if bits_stored and key == "0028|0101":
+            return str(bits_stored)
+        elif not monochrome2 and key == "0028|0004":
+            return "MONOCHROME1"
+        else:
+            return get_metadata_function(image_reader, key)
 
-    image_and_segmentation = load_image_in_known_formats(dcm_file, load_segmentation=False)
-    assert image_and_segmentation.images.ndim == 3 and image_and_segmentation.images.shape == (1,) + array_size
-    assert np.array_equal(image_and_segmentation.images, array[None, ...])
+    return mock_function
 
 
-def test_load_dicom_image_random_unsigned(test_output_dirs: TestOutputDirectories) -> None:
+@pytest.mark.parametrize("signed", [True, False])
+@pytest.mark.parametrize("monochrome2", [True, False])
+def test_load_dicom_image_ones(test_output_dirs: TestOutputDirectories,
```

Comment:
here and in related functions: I'd use "is_" as prefix for boolean variables (or has_... in other contexts)